test/Test/Database/LSMTree/StateMachine.hs

```diff
@@ -1970,8 +1956,14 @@ arbitraryActionWithVars _ label ctx (ModelState st _stats) =
           -- implemented
         , isJust (eqT @h @ModelIO.Table)
         ]
-     ++ [ (2,  fmap Some $ (Action <$> genErrors <*>) $
-            Unions <$> gen2or3TableVars)
+     ++ [ (2,  fmap Some $ (Action <$> genErrors <*>) $ do
+            -- Generate at least a 2-way union, and at most a 3-way union.
+            --
+            -- Tests for 0-way and 1-way unions are included in the UnitTests
+            -- module. n-way unions for n>3 lead to large unions, which are less
+            -- likely to be finished before the end of an action sequence.
+            n <- QC.chooseInt (2, 3)
+            Unions . NE.fromList <$> QC.vectorOf n genTableVar)
```

Comment:
Yeah this is an improvement over what `gen2or3TableVars` was doing :stuck_out_tongue: